docs/pages/access-controls/guides/guide-to-cluster-wide-idp-hardening.mdx

```diff
@@ -0,0 +1,62 @@
+
+## Current security measures may fall short
+
+While many organizations have implemented basic security measures like single sign-on (SSO) and multi-factor authentication (MFA), 
+these alone may not be sufficient to protect against sophisticated attacks targeting your IdP. 
+Attackers are constantly evolving their techniques, and traditional security measures may have limitations 
+or vulnerabilities that can be exploited.
+
+## Hardening measures
+
+To significantly enhance your defense against IdP compromises, we recommend implementing the following comprehensive security measures:
+
+### Set up cluster-wide WebAuthn
+Implement strong, phishing-resistant authentication across 
+your entire infrastructure using WebAuthn standards. WebAuthn, a W3C standard and part of FIDO2, 
+enables public-key cryptography for web authentication. Teleport supports WebAuthn as a second factor 
+for logging into Teleport (via tsh login or Web UI) and accessing SSH nodes or Kubernetes clusters. 
+It's compatible with hardware keys (e.g., YubiKeys, SoloKeys) and biometric authenticators like Touch ID and Windows Hello.
+
+### Establish cluster-wide per-session MFA
+Ensure that multi-factor authentication is required for each session, not just at initial login, 
+to maintain continuous security. Teleport's per-session MFA enhances security by protecting 
+against compromised on-disk certificates. It requires additional MFA checks when initiating new SSH, Kubernetes, database, or desktop sessions.
+
+### Implement cluster-wide device trust
+Develop a system to verify and manage trusted devices across your organization, reducing the risk 
+of unauthorized access from unknown or compromised devices. Device trust adds an extra layer of security by requiring the use of trusted devices 
+for accessing protected resources, complementing user identity and role enforcement. This can be 
+configured cluster-wide or via RBAC. Supported resources include apps (role-based only), SSH nodes, 
+databases, Kubernetes clusters, and first MFA device enrollment. The latter helps prevent auto-provisioning of new users through compromised IdPs.
+
+### Enable MFA for admin actions
+Add an extra layer of security for sensitive administrative operations by requiring multi-factor authentication 
+for these high-privilege actions. Teleport enforces additional MFA verification for administrative 
+actions across all clients (tctl, tsh, Web UI, and Connect). This feature adds an extra security 
+layer by re-verifying user identity immediately before any admin action, mitigating risks from compromised admin accounts.
+
+By adopting these advanced security measures, you can create a robust defense against IdP compromises and significantly reduce your organization's attack surface. 
+In the following sections, we'll dive deeper into each of these recommendations, providing step-by-step guidance on implementation and best practices.
+
+## Next steps
```

Comment:
I would like this guide to be self-contained, one-page instruction on how to enable those measures. An administrator looking at this page and following 4 steps above, should have fully protected system.

Now, this should also mention that the minimum requirement is Teleport 16. Please also test all instructions end to end.

Reply:
✔